include/swift/Basic/JSONSerialization.h

```diff
@@ -341,6 +342,17 @@ struct unvalidatedObjectTraits : public std::integral_constant<bool,
 && !has_ObjectValidateTraits<T>::value> {};
 
 class Output {
+public:
+  /// Enum of all Output subclasses to dynamically cast the output object and
+  /// thus pass serialization options specific to the serialization of a
+  /// specific object, e.g. to omit nodes from a syntax tree that have been
+  /// reused since the last serialization.
+  enum OutputKind {
+    Normal,
+    SyntaxTree
```

Comment:
It's still kind of weird that the serialization util knows syntax tree :( . Can we define a 32-bit field in `Output` whose meaning can be interpreted while serializing? For instance, we can use the first bit to indicate whether `shouldIncludeNodeIds`.